Obstacle ob_in
---
Obstacle ob_out
